url6f0c51b1b4_TheFakeMontyOnTheRun_open688_tgz-pJ8-br_odb_open688_app_Open688ServerJ8/src/br/odb/open688/app/net/TelnetClientServer.java

```diff
@@ -0,0 +1,133 @@
+package br.odb.open688.app.net;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import br.odb.gameapp.ApplicationClient;
+import br.odb.utils.FileServerDelegate;
+
+public class TelnetClientServer implements ApplicationClient {
+
+	volatile ServerSocket server;
+	volatile Socket socket;
+	public volatile DataOutputStream out;
+	public volatile DataInputStream din;
+	public final int clientId;
+	int port;
+	
+	public TelnetClientServer( int clientId ) {
+		this.clientId = clientId;
+		port = 2337 + clientId;
+	}
+	
+	public void openServer() {
+		try {
+			server = new ServerSocket( port );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *server*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).